exporter/opentelemetry-exporter-jaeger/src/opentelemetry/exporter/jaeger/__init__.py

```diff
@@ -194,6 +191,12 @@ def __init__(
             if transport_format
             else TRANSPORT_FORMAT_THRIFT
         )
+        tracer_provider = trace.get_tracer_provider()
+        self.service_name = (
+            tracer_provider.resource.attributes[SERVICE_NAME]
+            if getattr(tracer_provider, "resource", None)
+            else Resource.create({})
```

Comment:
Is setting the `service_name` to a new `Resource` if `tracer_provider` does not have a `resource` attribute?

Reply:
But the `service_name` should be a string right? Not a new `Resource`?

Reply:
This Resource.create({}) defaults to "unknown_service" (a string) if it doesnt find service_name as part of env var.

Reply:
> Is setting the `service_name` to a new `Resource` if `tracer_provider` does not have a `resource` attribute?

This scenario is possible only if DefaultTracerProvider (from opentelemetry api) is instantiated.   (DefaultTracerProvider  does not have resource attribute) 
The TracerProvider from sdk has resource attribute which is set to default value of Resource.create({}).

Resource.create({}) sets the service_name to "unknown_service" if it doesnt find the env var OTEL_RESOURCE_ATTRIBUTES.

Reply:
Looks like you are confused a little `service_name` should be string type not an instance of `Resource`. If a tracer_provider doesn't have resource attribute you are setting `Resource` instance to service_name which is not expected. You probably meant something like `Resource.create().attributes[SERVICE_NAME]`. Anyway as we are going to change the approach and will remove this but just pointing out the issue here.

Reply:
@lonewolf3739 
As pointed out above, your scenario is perfectly valid, I was just referring to setting the variable to the actual default service name instead of the `Resource` instance.

Also, even though we might be changing the approach, we should move ahead with these changes for now and leave consistency changes for a different PR.

Reply:
the change as been done